syft/pkg/cataloger/terraform/parse_tf_lock.go

```diff
@@ -0,0 +1,55 @@
+package terraform
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/hashicorp/hcl/v2/hclsimple"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+type terraformLockFile struct {
+	Providers []pkg.TerraformLockEntry `hcl:"provider,block"`
+}
+
+func parseTerraformLock(_ context.Context, _ file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	var lockFile terraformLockFile
+
+	contents, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read terraform lock file: %w", err)
+	}
+
+	err = hclsimple.Decode(reader.RealPath, contents, nil, &lockFile)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode terraform lock file: %w", err)
+	}
+
+	pkgs := make([]pkg.Package, 0, len(lockFile.Providers))
+
+	for _, provider := range lockFile.Providers {
+		pkg := pkg.Package{
+			Name:      provider.URL,
+			Version:   provider.Version,
+			FoundBy:   "terraform-cataloger",
+			Locations: file.NewLocationSet(reader.Location.WithAnnotation(pkg.EvidenceAnnotationKey, pkg.PrimaryEvidenceAnnotation)),
+			Licenses:  pkg.NewLicenseSet(), // TODO: license could be found in .terraform/providers/${name}/${version}/${arch}/LICENSE.txt
+			// TODO: Language?
```

Comment:
technically providers are written in go... is this just because it's easier to make a provider in go (since the SDK is provided officially only for go)? or are there unofficial non-go options available? I feel like this could use `pkg.Golang` that we have today, but curious for thoughts here.

Reply:
See also: https://developer.hashicorp.com/terraform/plugin/best-practices/provider-code#other-languages

TL;DR: It is possible to write providers in other languages but there is no official SDK or even effort from Hashicorp to support you doing so.

So I'd say it is safe to say the language is `pkg.Golang`